File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/primitive_ref_runme.lua
require("import")	-- the import fn
import("primitive_ref")	-- import code
pr=primitive_ref --alias

assert(pr.ref_int(3)==3)

assert(pr.ref_uint(3) == 3)

assert(pr.ref_short(3) == 3)

assert(pr.ref_ushort(3) == 3)

assert(pr.ref_long(3) == 3)

assert(pr.ref_ulong(3) == 3)

assert(pr.ref_schar(3) == 3)

assert(pr.ref_uchar(3) == 3)

assert(pr.ref_float(3.5) == 3.5)

assert(pr.ref_double(3.5) == 3.5)

assert(pr.ref_bool(true) == true)

assert(pr.ref_char('x') == 'x')

assert(pr.ref_over(0) == 0)

a=pr.A(12)
assert(pr.ref_over(a)==12)

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/primitive_ref_runme.go
package main

import . "./primitive_ref"

func main() {
	if Ref_int(3) != 3 {
		panic(0)
	}

	if Ref_uint(3) != 3 {
		panic(0)
	}

	if Ref_short(3) != 3 {
		panic(0)
	}

	if Ref_ushort(3) != 3 {
		panic(0)
	}

	if Ref_long(3) != 3 {
		panic(0)
	}

	if Ref_ulong(3) != 3 {
		panic(0)
	}

	if Ref_schar(3) != 3 {
		panic(0)
	}

	if Ref_uchar(3) != 3 {
		panic(0)
	}

	if Ref_float(3.5) != 3.5 {
		panic(0)
	}

	if Ref_double(3.5) != 3.5 {
		panic(0)
	}

	if Ref_bool(true) != true {
		panic(0)
	}

	if Ref_char('x') != 'x' {
		panic(0)
	}

	if Ref_over(0) != 0 {
		panic(0)
	}
}
